From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harmer <mark.harmer@discordapp.com>
Date: Fri, 29 Mar 2024 12:23:03 -0700
Subject: fix: nvidia HDR crash on Windows 32-bit

Disable nvidia HDR on Windows 32-bit, causes GPU process to crash.

diff --git a/ui/gl/swap_chain_presenter.cc b/ui/gl/swap_chain_presenter.cc
index d0fe762f3b46fb8a4dcaa03d9df663ff542acf8b..f990ae884622f2f711606d39b8bbf2d03adea709 100644
--- a/ui/gl/swap_chain_presenter.cc
+++ b/ui/gl/swap_chain_presenter.cc
@@ -7,6 +7,7 @@
 #include <d3d11_1.h>
 #include <d3d11_4.h>
 
+#include "build/build_config.h"
 #include "base/debug/alias.h"
 #include "base/debug/crash_logging.h"
 #include "base/debug/dump_without_crashing.h"
@@ -383,10 +384,13 @@ HRESULT ToggleVpAutoHDR(UINT gpu_vendor_id,
 }
 
 bool IsVpAutoHDREnabled(UINT gpu_vendor_id) {
+  // NVidia HDR is broken on 32-bit builds
+#if !defined(ARCH_CPU_X86)
   if (gpu_vendor_id == 0x10de &&
       base::FeatureList::IsEnabled(features::kNvidiaVpTrueHDR)) {
     return true;
   }
+#endif
 
   return false;
 }
